types/react/test/hooks.tsx

```diff
@@ -270,18 +276,31 @@ function useEveryHook(ref: React.Ref<{ id: number }> | undefined): () => boolean
     // make sure setState accepts a function
     setToggle(r => !r);
 
-    // Undesired
-    // Should not type-check since `number` will be `number` at runtime but `() => number` in the type-checker
-    const [number, setNumber] = React.useState<() => number>(() => 0);
-    // Should be `number`
-    // $EpectType () => number
+    // $ExpectType number
+    React.useState(() => 0)[0];
+    // $ExpectType string
+    React.useState(() => "")[0];
+
+    React.useState();
```

Comment:
Can you rearrange the test so that it's clear if this is a breaking change or not? The diff is all over the place so it's hard to see what changed for existing test and what is a new test.

Ideally new tests have a separate commit with their current behavior and a second commit includes changes to types and the behavior change it causes.